### PR TITLE
intel_adsp: cleanup interrupt controller, timer and headers in general

### PR DESCRIPTION
**Pull request by nashif**
Cleanup headers for ACE and split them. General namespace improvements and removal of unused/duplicate code.

### CODE REVIEW

**Review by andyross — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

**Review by kv2019i — COMMENTED** [review]
Looks good to me. @mwasko @aborisovich @marcinszkudlinski  please have a look!

**Review by edersondisouza — CHANGES_REQUESTED** [review]
Minor comments, but it seems there's a missing file rename.

**Review by dcpleung — APPROVED** [review]

File: soc/xtensa/intel_adsp/ace/multiprocessing.c [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #include <adsp_memory.h>
 
 #define CORE_POWER_CHECK_NUM 32
+#define ACE_INTC_IRQ DT_IRQN(DT_NODELABEL(ace_intc))
```
Comment (edersondisouza) [inline soc/xtensa/intel_adsp/ace/multiprocessing.c:18]: Isn't this already defined at `ace_v1x-regs.h/adsp_interrupt.h`?

File: soc/xtensa/intel_adsp/cavs/include/intel_tgl_adsp/adsp_shim.h [hunk 2]
```diff
@@ -27,41 +27,27 @@ struct cavs_shim {
...
+	uint32_t lpsalhss2; /* Offset: 0x00B8 */
+	uint32_t lpsalhss3; /* Offset: 0x00BC */
 	uint32_t _unused5[4];
-	uint32_t l2mecs;
-	uint32_t l2mpat;
-	uint32_t _unused6[2];
-	uint32_t ltrc;
+	uint32_t l2mecs;  /* Offset: 0x00D0 */
+	uint32_t _unused6;
+	uint32_t _unused7[2];
```
Comment (edersondisouza) [inline soc/xtensa/intel_adsp/cavs/include/intel_tgl_adsp/adsp_shim.h:46]: Why not merge this two "unused"?

**Review by edersondisouza — APPROVED** [review]
Minor nitpicks only

**Pull request merged**